### ISSUE TITLE
🐛 BUG: Scoped component styling seems to be broken in Astro v0.21.2

### ISSUE THREAD

**Issue by markteekman** [withastro/astro#1992]
### What version of `astro` are you using?

0.21.2

### What package manager are you using?

npm

### What operating system are you using?

Mac

### Describe the Bug

I was migrating https://github.com/markteekman/accessible-astro-starter to Astro v0.21.2. I noticed some strange behavior when running `npm start` after installing the newest version of Astro. It seems scoped styling isn't applied to components anymore. For example, I've got this components HTML:

```html
<header>
  <SkipLinks client:load />
  <section class="padding-16 color-neutral">
    <div class="container">
      <img src="/assets/logo.svg" alt="Your Logo">
      <ResponsiveToggle client:load />
      <Navigation client:load />
    </div>
  </section>
</header>
```

And the following SCSS:

```scss
<style lang="scss">
  @use '../styles/base/breakpoint' as *;

  .container {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: space-between;
    align-items: center;
    gap: 1rem;

    @include breakpoint(medium) {
      flex-wrap: nowrap;
    }

    img {
      max-height: 40px;
    }
  }
</style>
```

All styles on the `.container` class are not applied when viewing the website. Only when I use `<style lang="scss" global>`, these styles are applied. But that's obvious I guess, since it's no longer scoped to the component 🙂 

Solved for now by setting `global` to the necessary style tags, but this also required adding an extra `.temp` class because now it would effect ALL `.container` classes. I tried reproducing this on the Astro Starter on StackBlitz, but it seems to be working just fine, which makes it hard for me to debug and determine where it is coming from.. In the example I've added a container class, made the background green and set the style tag to `langauge="scss"`.

I did update my Accessible Astro Starter theme using the global tag, so you can also install it and test it there with `npm init astro -- --template markteekman/accessible-astro-starter` and then remove the global tag and the `.temp` class. 🙂 

Thanks in advance!

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-symmkd?file=src/components/Tour.astro

**Comment by matthewp** [comment]
Thanks for the reproduction. Obviously you are right, these should be scoped, so we'll have to dig into what is going on.

**Comment by drwpow** [comment]
This is what I’m seeing locally:

```html
<style>
/*
 | -------------------------------------------------------------
 |	--> breakpoint
 | -------------------------------------------------------------
 |
 | 	Breakpoint mixin used in the other base files.
 |
*/;.container.astro-UGMBVI4R{display:flex;flex-direction:row;flex-wrap:wrap;justify-content:space-between;align-items:center;gap:1rem;}@media(min-width:48em){.container.astro-UGMBVI4R{flex-wrap:nowrap;}}.container.astro-UGMBVI4R img.astro-UGMBVI4R{max-height:40px;}
</style>
```

For some reason, the `@use` is inserting that comment, along with a semicolon after it. Not sure if this is a bug within our compiler or what, but if that `;` isn’t there it’ll work fine.

This is definitely a bug, but in the meantime one workaround is to use `//` comments in Sass that don’t get rendered to the final CSS.

**Comment by markteekman** [comment]
Thanks for looking in to this so quicly @matthewp and @drwpow! I thought I'd tested it with removing that @use statement, but apparently not.. 🤔 Thanks for that. Going with regular `//` comments might even be considered an improvement rather then a workaround in this case!

**Comment by drwpow** [comment]
Update: this has been fixed in the latest compiler and should be fixed in `0.21.4`. Thanks for filing this! I’m sure this was probably affecting a lot of users.

**Comment by markteekman** [comment]
Thanks @drwpow and thank you for fixing this so soon 😊
